### PR TITLE
RF: small streamlines API PR edits

### PR DESCRIPTION
**Pull request by matthew-brett**
A little refactoring of the Trk file `_read_header` method to save an
unnecessary seek.

Some rewriting and expansion of docstrings.

### CODE REVIEW

File: nibabel/streamlines/tests/test_trk.py [hunk 1]
```diff
@@ -460,3 +449,47 @@ def test_write_scalars_and_properties_name_too_long(self):
...
+            bio.write(fobj.read())
+        bio.seek(hdr_pos)
+        # Check header is as expected
+        hdr_from_fname['_offset_data'] += hdr_pos  # Correct for start position
+        assert_arr_dict_equal(TrkFile._read_header(bio), hdr_from_fname)
+        # Check fileobject file position has not changed
+        assert_equal(bio.tell(), hdr_pos)
+
+
+def test_encode_names():
```
Comment (MarcCote) [inline nibabel/streamlines/tests/test_trk.py:470]: Oops, totally forgot to add those unit tests. Thanks.

File: nibabel/streamlines/trk.py [hunk 2]
```diff
@@ -88,6 +88,24 @@
 
 
 def get_affine_trackvis_to_rasmm(header):
+    """ Get affine mapping trackvis voxelmm space to RAS+ mm space
+
+    The streamlines in a trackvis file are in 'voxelmm' space, where the
+    coordinates refer to the corner of the voxel.
+
+    Compute the # affine matrix that will bring them back to RAS+ mm space,
+    where the coordinates refer to the center of the voxel.
```
Comment (MarcCote) [inline nibabel/streamlines/trk.py:97]: I like the wording `where the coordinates refer to the center of the voxel` :).

File: nibabel/streamlines/trk.py [hunk 3]
```diff
@@ -88,6 +88,24 @@
 
 
 def get_affine_trackvis_to_rasmm(header):
+    """ Get affine mapping trackvis voxelmm space to RAS+ mm space
+
+    The streamlines in a trackvis file are in 'voxelmm' space, where the
+    coordinates refer to the corner of the voxel.
+
+    Compute the # affine matrix that will bring them back to RAS+ mm space,
```
Comment (MarcCote) [inline nibabel/streamlines/trk.py:96]: Has `#` some special meaning here or it is a typo?

File: nibabel/streamlines/trk.py [hunk 4]
```diff
@@ -388,13 +404,13 @@ def _read():
         return cls(tractogram, header=hdr)
 
     def save(self, fileobj):
-        """ Saves tractogram to a file-like object using TRK format.
+        """ Save tractogram to a filename or file-like object using TRK format.
```
Comment (MarcCote) [inline nibabel/streamlines/trk.py:407]: I thought the verb was supposed to be conjugated in third-person singular when describing what a function does?

**Comment by MarcCote** [comment]
Really like the refactoring you've done. Also, thanks a lot for fixing the docstring.
I'll merge it right now as the comments do not require real changes.

File: nibabel/streamlines/trk.py (hunk 3 above)
Comment (matthew-brett) [inline nibabel/streamlines/trk.py:96]: The # is a typo - sorry.

File: nibabel/streamlines/trk.py (hunk 4 above)
Comment (matthew-brett) [inline nibabel/streamlines/trk.py:407]: > I thought the verb was supposed to be conjugated in third-person singular when describing what a function does?

It's not a big deal either way, and I'm inconsistent in doing this,
but I was using the imperative (I think) - there's an example from the
longer PEP document :
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

```
The docstring is a phrase ending in a period. It prescribes the
function or method's effect as a command ("Do this", "Return that"),
not as a description; e.g. don't write "Returns the pathname ...".
```

File: nibabel/streamlines/trk.py (hunk 3 above)
Comment (matthew-brett) [inline nibabel/streamlines/trk.py:96]: I fixed the typo in #10.

**Pull request merged**